File: src/main/typescript/wcc/graphic/graphic-component.ts
import { Graphic } from "../../db/graphic";
import { GraphicFound } from "../../db/graphic-found";
import { GraphicPieceComponent } from "./graphic-piece-component";

export interface GraphicComponent {
	readonly piece: GraphicPieceComponent;

	search(word: string): Promise<GraphicFound[]>;
	get(id: number): Promise<Graphic>;
	getByName(name: string): Promise<Graphic>;
	save(graphic: Graphic): Promise<number>;
	delete(id: number): Promise<void>;
}

File: src/main/java/app/wcc/graphic/GraphicComponent.java
package app.wcc.graphic;

import org.wcardinal.controller.annotation.Callable;
import org.wcardinal.controller.annotation.Component;

import app.db.graphic.Graphic;
import app.db.graphic.GraphicFound;
import app.db.graphic.GraphicService;

import org.springframework.beans.factory.annotation.Autowired;

@Component
public class GraphicComponent {
	@Autowired
	GraphicService service;

	@Autowired
	GraphicPieceComponent piece;

	@Callable
	Iterable<GraphicFound> search(String word) {
		return service.search(word);
	}

	@Callable
	Graphic get(final Long id) {
		return service.find(id).orElse(null);
	}

	@Callable
	Graphic getByName(final String name) {
		return service.find(name).orElse(null);
	}

	@Callable
	Long save(final Graphic graphic) {
		return service.save(graphic).getId();
	}

	@Callable
	void delete(Long id) {
		service.delete(id);
	}
}
